kernel-impl/src/main/java/org/sakaiproject/content/impl/CollectionAccessFormatter.java

```diff
@@ -214,8 +214,10 @@ public void format(ContentCollection x, Reference ref, HttpServletRequest req, H
 					URI contentUri = new URI(contentUrl);
 					URI relativeUri = baseUri.relativize(contentUri);
 					contentUrl = relativeUri.toString();
+					if(content.isHidden()) {
```

Comment:
I'd call `isAvailable()` again, otherwise if you're in a folder that's not hidden while iterating though the contents you come across a folder that is time restricted it won't get greyed out. `isAvailable()` is a little slower and does more checks that we know to be useless but I think it's sensible.

Otherwise you end up that when your view a time limited folder it will be greyed out, but when you see it in the parent folder listing it won't be.

I know time limiting was outside the original Jira but I think it's sensible to make the fix consistent, rather than introducing another edge case.